app/services/sdr/event.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Sdr
+  # Service to interact with SDR Event Service.
+  class Event
+    class Error < StandardError; end
+
+    # @param [String] druid
+    # @param [String] type the type of event
+    # @param [Hash] data the event data
+    # # @raise [Error] if there is an error creating the event
+    def self.create(druid:, type:, data:)
+      Dor::Event::Client.create(druid:, type:, data:) if Settings.rabbitmq.enabled
+    rescue StandardError => e
+      raise Error, "Failed to create event for #{druid}: #{e.message}"
```

Comment:
Is there any other context in `e` that we care about other than the message? E.g., the class or the backtrace? If so, we may want to include that in the re-raised exception.

Reply:
I don't think so. In other code bases, we don't even rescue when calling `.create`. And Dor::Event::Client doesn't do any error handling. If we ever actually encounter errors, I think better error handling should be added to the client.